File: solutions/javascript/bird-watcher/1/bird-watcher.js
// @ts-check
//
// The line above enables type checking for this file. Various IDEs interpret
// the @ts-check directive. It will give you helpful autocompletion when
// implementing this exercise.

/**
 * Calculates the total bird count.
 *
 * @param {number[]} birdsPerDay
 * @returns {number} total bird count
 */
export function totalBirdCount(birdsPerDay) {
  let numberOfBirds = 0
  for (let name of birdsPerDay) {
    numberOfBirds += name
  }
  return numberOfBirds 
}

/**
 * Calculates the total number of birds seen in a specific week.
 *
 * @param {number[]} birdsPerDay
 * @param {number} week
 * @returns {number} birds counted in the given week
 */
export function birdsInWeek(birdsPerDay, week) {
  let days= 7
  let startOfDay = (week - 1) * days
  let endOfDay = week * days
  let birds = 0
   for (let index = startOfDay; index <= endOfDay -1; index++) {
     birds += birdsPerDay[index]
   }
   return birds
}

/**
 * Fixes the counting mistake by increasing the bird count
 * by one for every second day.
 *
 * @param {number[]} birdsPerDay
 * @returns {number[]} corrected bird count data
 */
export function fixBirdCountLog(birdsPerDay) {
  for (let index = 0; index < birdsPerDay.length; index++) {
    if(index % 2 == 0){
      birdsPerDay[index]++
    }
  } 
  return birdsPerDay
}

File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
    var value int = 0
	for _,i := range birdsPerDay{
        value += i
    }
    return value
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
    return TotalBirdCount(birdsPerDay[((week - 1)*7):(week * 7)]) 
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for i:= 0; i < len(birdsPerDay); i++ {
        if i&1 != 0{
            continue
        }
        birdsPerDay[i]+=1
    }
    return birdsPerDay
}
